src/main/java/org/opensearch/neuralsearch/search/CompoundTopDocs.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.search;
+
+import java.util.Arrays;
+
+import lombok.Getter;
+import lombok.ToString;
+
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.TotalHits;
+
+/**
+ * Class stores collection of TodDocs for each sub query from hybrid query
+ */
+@ToString(includeFieldNames = true)
+public class CompoundTopDocs extends TopDocs {
+
+    @Getter
+    private TopDocs[] compoundTopDocs;
+
+    public CompoundTopDocs(TotalHits totalHits, ScoreDoc[] scoreDocs) {
+        super(totalHits, scoreDocs);
+    }
+
+    public CompoundTopDocs(TotalHits totalHits, TopDocs[] docs) {
+        // we pass clone of score docs from the sub-query that has most hits
```

Comment:
problem they are facing is that TopDocs is used by core classes and we can't extend or change those classes. So we're keeping TopDocs up to the contract and also add out specific information that we'll use in later stages from pipeline post processor. Constructor ```CompoundTopDocs(TotalHits totalHits, ScoreDoc[] scoreDocs)``` is existing core super constructor, we can't change it. ```CompoundTopDocs(TotalHits totalHits, TopDocs[] docs)``` is what we added to have sub-queries results. 

As per core contract we can have only one set of results, and only one query. In case of hybrid search some sub-queries at shard level may have no hits, so if we pick those sub-query results and use them it may influence later stages (in case they do have some optimizations to drop such shard results). Picking one with most results (any in there is a tie) looks more relevant for hybrid queries, as after normalization and combination we'll get max (total_hits) results. 

For example:

shard level results:
```
sub-query_1 {doc2: 1.0},
sub-query_2 {doc1: 0.4, doc2: 2.0},
sub_query_3 {}
```
we build CompoundTopDocs
```
{doc1: 0.4, doc2: 2.0}, total_hits: 2
```

after scores combination:
```
{doc2: 3.0, doc1: 0.4} 
```

resulting list looks much more like sub_query results with max hits.

Reply:
What happen in this case? Will it still return two results?
```
sub-query_1 {doc2: 1.0},
sub-query_2 {doc1: 0.4},
sub_query_3 {}
```

Reply:
In this case CompoundTopDocs will have:
```
ScoreDocs: {doc2: 1.0}, totalHits = 1

TopDocs: 
{doc2: 1.0},
{doc1: 0.4},
{}
```
then after normalization, if  query has ```size=1```  then we will return:
```
{doc2: 1.0}
```
because it has the highest score, if ```size >= 2``` then return
```
{doc2: 1.0},
{doc1: 0.4}
```